diff -ur xkeyboard-config-2.27/rules/base.xml xkeyboard-config-2.27/rules/base.xml
--- xkeyboard-config-2.27/rules/base.xml	2019-05-30 23:30:31.000000000 +0000
+++ xkeyboard-config-2.27/rules/base.xml	2019-10-13 11:32:35.693430368 +0000
@@ -4595,6 +4595,12 @@
         </variant>
         <variant>
           <configItem>
+             <name>dvp</name>
+             <description>Russian (Programmer Dvorak compatible)</description>
+           </configItem>
+         </variant>
+         <variant>
+           <configItem>
             <name>phonetic_winkeys</name>
             <description>Russian (phonetic, with Win keys)</description>
           </configItem>
diff -ur xkeyboard-config-2.27/rules/evdev.xml xkeyboard-config-2.27/rules/evdev.xml
--- xkeyboard-config-2.27/rules/evdev.xml	2019-05-30 23:30:43.000000000 +0000
+++ xkeyboard-config-2.27/rules/evdev.xml	2019-10-13 11:33:27.451759244 +0000
@@ -4595,6 +4595,12 @@
         </variant>
         <variant>
           <configItem>
+             <name>dvp</name>
+             <description>Russian (Programmer Dvorak compatible)</description>
+           </configItem>
+         </variant>
+         <variant>
+           <configItem>
             <name>phonetic_winkeys</name>
             <description>Russian (phonetic, with Win keys)</description>
           </configItem>
diff -ur xkeyboard-config-2.27/symbols/ru xkeyboard-config-2.27/symbols/ru
--- xkeyboard-config-2.27/symbols/ru	2019-10-13 11:19:08.000000000 +0000
+++ xkeyboard-config-2.27/symbols/ru	2019-10-13 11:29:51.891193005 +0000
@@ -856,6 +856,28 @@
 	include "sun_vndr/ru(sun_type6)"
 };
 
+partial default alphanumeric_keys
+xkb_symbols "dvp" {
+    include "ru(common)"
+
+    name[Group1]= "Russia";
+
+    key <AE01> {        [       numerosign, percent   ]       };
+    key <AE02> {        [       quotedbl,   7         ]       };
+    key <AE03> {        [       question,   5         ]       };
+    key <AE04> {        [       slash,      3         ]       };
+    key <AE05> {        [       parenleft,  1         ]       };
+    key <AE06> {        [       equal,      9         ]       };
+    key <AE07> {        [       asterisk,   0         ]       };
+    key <AE08> {        [       parenright, 2         ]       };
+    key <AE09> {        [       plus,       4         ]       };
+    key <AE10> {        [       minus,      6         ]       };
+    key <AE11> {        [       exclam,     8         ]       };
+    key <AE12> {        [       semicolon,  colon     ]       };
+    key <AB10> {        [       period,     comma     ]       };
+    key <BKSL> {        [       backslash,  bar       ]       };
+};
+
 // Apple layout
 partial alphanumeric_keys
 xkb_symbols "mac" {
