diff -ru xkeyboard-config-2.20.orig/rules/base.xml.in xkeyboard-config-2.20/rules/base.xml.in
--- xkeyboard-config-2.20.orig/rules/base.xml.in	2017-02-01 04:39:10.000000000 +0300
+++ xkeyboard-config-2.20/rules/base.xml.in	2017-03-20 18:20:11.338387281 +0300
@@ -4092,6 +4092,12 @@
         </variant>
         <variant>
           <configItem>
+            <name>dvp</name>
+            <_description>Russian (Programmer Dvorak compatible)</_description>
+          </configItem>
+        </variant>
+        <variant>
+          <configItem>
             <name>phonetic_winkeys</name>
             <_description>Russian (phonetic WinKeys)</_description>
           </configItem>
diff -ru xkeyboard-config-2.20.orig/rules/evdev.xml.in xkeyboard-config-2.20/rules/evdev.xml.in
--- xkeyboard-config-2.20.orig/rules/evdev.xml.in	2017-02-01 04:39:18.000000000 +0300
+++ xkeyboard-config-2.20/rules/evdev.xml.in	2017-03-20 18:20:40.302719956 +0300
@@ -4092,6 +4092,12 @@
         </variant>
         <variant>
           <configItem>
+            <name>dvp</name>
+            <_description>Russian (Programmer Dvorak compatible)</_description>
+          </configItem>
+        </variant>
+        <variant>
+          <configItem>
             <name>phonetic_winkeys</name>
             <_description>Russian (phonetic WinKeys)</_description>
           </configItem>
diff -ru xkeyboard-config-2.20.orig/symbols/ru xkeyboard-config-2.20/symbols/ru
--- xkeyboard-config-2.20.orig/symbols/ru	2017-02-01 04:39:10.000000000 +0300
+++ xkeyboard-config-2.20/symbols/ru	2017-03-20 18:16:20.425578176 +0300
@@ -853,3 +853,25 @@
 
     include "level3(ralt_switch)"
 };
+
+partial default alphanumeric_keys
+xkb_symbols "dvp" {
+    include "ru(common)"
+
+    name[Group1]= "Russia";
+
+    key <AE01> {        [       numerosign, percent   ]       };
+    key <AE02> {        [       quotedbl,   7         ]       };
+    key <AE03> {        [       question,   5         ]       };
+    key <AE04> {        [       slash,      3         ]       };
+    key <AE05> {        [       parenleft,  1         ]       };
+    key <AE06> {        [       equal,      9         ]       };
+    key <AE07> {        [       asterisk,   0         ]       };
+    key <AE08> {        [       parenright, 2         ]       };
+    key <AE09> {        [       plus,       4         ]       };
+    key <AE10> {        [       minus,      6         ]       };
+    key <AE11> {        [       exclam,     8         ]       };
+    key <AE12> {        [       semicolon,  colon     ]       };
+    key <AB10> {        [       period,     comma     ]       };
+    key <BKSL> {        [       backslash,  bar       ]       };
+};
