--- emacs/src/pgtkselect.c-	2022-01-30 00:32:53.834311402 +0900
+++ emacs/src/pgtkselect.c	2022-01-30 00:36:37.514307007 +0900
@@ -347,6 +347,9 @@
   if (!pgtk_selection_usable ())
     return Qnil;

+  if (!EQ(selection, QCLIPBOARD))
+    return Qnil;
+
   if (NILP (frame))
     frame = selected_frame;
   if (!FRAME_LIVE_P (XFRAME (frame)) || !FRAME_PGTK_P (XFRAME (frame)))
@@ -437,6 +440,9 @@
   if (!f)
     return Qnil;

+  if (!EQ(selection, QCLIPBOARD))
+    return Qnil;
+
   cb = symbol_to_gtk_clipboard (FRAME_GTK_WIDGET (f), selection);

   gtk_clipboard_clear (cb);
@@ -468,6 +474,9 @@
   if (!f)
     return Qnil;

+  if (!EQ(selection, QCLIPBOARD))
+    return Qnil;
+
   cb = symbol_to_gtk_clipboard (FRAME_GTK_WIDGET (f), selection);

   return gtk_clipboard_wait_is_text_available (cb) ? Qt : Qnil;
@@ -497,6 +506,9 @@
   if (!pgtk_selection_usable ())
     return Qnil;

+  if (!EQ(selection, QCLIPBOARD))
+    return Qnil;
+
   cb = symbol_to_gtk_clipboard (FRAME_GTK_WIDGET (f), selection);
   selection_type_to_quarks (gtk_clipboard_get_selection (cb), &quark_data,
 			    &quark_size);
@@ -538,6 +550,9 @@
   if (!pgtk_selection_usable ())
     return Qnil;

+  if (!EQ(selection_symbol, QCLIPBOARD))
+    return Qnil;
+
   cb = symbol_to_gtk_clipboard (FRAME_GTK_WIDGET (f), selection_symbol);

   GdkAtom target_atom = gdk_atom_intern (SSDATA (SYMBOL_NAME (target_type)), false);
