diff --git a/ddcci/ddcci.c b/ddcci/ddcci.c
index 13ddb85..cf70c1e 100644
--- a/ddcci/ddcci.c
+++ b/ddcci/ddcci.c
@@ -464,7 +464,7 @@ static int ddcci_identify_device(struct i2c_client *client, unsigned char addr,
 	}
 
 	/* validate first byte */
-	if (buffer[0] != addr) {
+	if (buffer[0] != addr && (addr != 0x6e && buffer[0] != 0x6f)) {
 		dev_dbg(&client->dev,
 			"[%02x:%02x] identification response: %*ph\n",
 			client->addr << 1, addr, (ret > 32 ? 32 : ret), buffer);
